client/tck/src/main/java/tck/graphql/dynamic/core/BuildDirectiveTest.java

```diff
@@ -0,0 +1,342 @@
+package tck.graphql.dynamic.core;
+
+import static io.smallrye.graphql.client.core.Argument.arg;
+import static io.smallrye.graphql.client.core.Directive.directive;
+import static io.smallrye.graphql.client.core.DirectiveArgument.directiveArg;
+import static io.smallrye.graphql.client.core.Document.document;
+import static io.smallrye.graphql.client.core.Field.field;
+import static io.smallrye.graphql.client.core.Field.fieldWithDirectives;
+import static io.smallrye.graphql.client.core.InlineFragment.on;
+import static io.smallrye.graphql.client.core.Operation.operation;
+import static io.smallrye.graphql.client.core.Operation.operationWithDirectives;
+import static io.smallrye.graphql.client.core.OperationType.MUTATION;
+import static io.smallrye.graphql.client.core.OperationType.QUERY;
+import static io.smallrye.graphql.client.core.OperationType.SUBSCRIPTION;
+import static io.smallrye.graphql.client.core.ScalarType.GQL_ID;
+import static io.smallrye.graphql.client.core.ScalarType.GQL_INT;
+import static io.smallrye.graphql.client.core.Variable.var;
+import static io.smallrye.graphql.client.core.Variable.varWithDirectives;
+import static io.smallrye.graphql.client.core.VariableType.nonNull;
+import static io.smallrye.graphql.client.core.VariableType.varType;
+
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+
+import io.smallrye.graphql.client.core.Document;
+import io.smallrye.graphql.client.core.Variable;
+import tck.graphql.dynamic.helper.AssertGraphQL;
+import tck.graphql.dynamic.helper.Utils;
+
+/**
+ * This class tests creating queries (via DSL) that contain user-made client-side directives (with/without arguments).
+ */
+public class BuildDirectiveTest {
+    /*--------TESTING INLINE FRAGMENTS--------*/
+    @Test
+    public void buildInlineFragmentWithDirectiveTest() {
+        String expectedRequest = Utils.getResourceFileContent("core/directivesInlineFragment.graphql");
+        Document document = document(
+                operation(QUERY,
+                        "SimpleQuery",
+                        field("customers",
+                                field("name"),
+                                on("Customer",
+                                        List.of(
```

Comment:
use `Directive.directives()` instead of `List.of` to create a list, it's the recommended way to use the DSL